### ISSUE TITLE
Clearing a form after submission

### ISSUE THREAD

**Issue by duro** [offirgolan/ember-cp-validations#228]
I have a form that allows a user to create a model. This form is created as a component that open in a panel on button click. Upon successful creation the panel closes. If the panel is opened back up, the form should be empty, ready for a new entry. This is not the case. Even if the form panel component is destroyed and re-init'd, it seems the form values are still there.

I have tried setting the model that the form is using back to null or {}, but the problem here is that it triggers all the validations and leaves me with the error messages.

So I guess my problem is two fold. One, the form values persist even beyond component destruction and re-init. And problem two, there is no clear way to reset the model that the form uses.

I am using `ember-bootstrap` and `ember-bootstrap-cp-validations`.

This is the component logic:

```
import Ember from 'ember';
import { validator, buildValidations } from 'ember-cp-validations';

const Validations = buildValidations({
  'formData.appName': {
    description: 'App Name',
    validators: [
      validator('presence', true)
    ]
  },
  'formData.url': {
    description: 'URL',
    validators: [
      validator('presence', true),
      validator('format', {
        type: 'url',
        description: 'The URL'
      })
    ]
  }
});

export default Ember.Component.extend(Validations, {
  classNameBindings: ['styles.container'],
  formData: {},
  actions: {
    handleSubmit() {
      const formData = this.get('formData');
      const onSubmit = this.get('onSubmit');
      const currentFolder = this.get('currentFolder');

      onSubmit(formData, currentFolder)
        .then((/* savedModel */) => {
          this.set('displaySuccess', true);
          this.set('formData', {});
          Ember.run.later(() => {
            this.get('onClose')();
          }, 1000);
        })
        .catch(err => {
          console.log('Save Error', err);
        });
    }
  }
});
```

And here is the component template:

```
<h3>Add New External App</h3>

{{#if displaySuccess}}
<p class={{styles.success}}>New App Created!</p>
{{/if}}

{{#bs-form model=this action='handleSubmit'}}
  {{bs-form-element controlType="text" label="App Name" placeholder="App Name" property="formData.appName" name="title" class="title-group"}}
  {{bs-form-element controlType="text" label="URL" placeholder="http://www.example.com" property="formData.url" name="url" class="url-group"}}
  {{bs-button defaultText="Submit" type="primary" buttonType="submit" class="submitButton"}}
{{/bs-form}}
```

Any help someone can provide is greatly appreciate. I've been tearing my hair out for hours on this, and getting nowhere.

**Comment by offirgolan** [comment]
How I see it, you have a couple options. 

1. You can use an addon to help manage the state of your `formData` either by using [ember-time-machine](https://github.com/offirgolan/ember-time-machine) or via [ember-buffered-proxy](https://github.com/yapplabs/ember-buffered-proxy)

2. You can place your validations on your model instead of the component. I dont really know why its not being properly destroyed tbh... maybe its a bug with ember-cp-validations but I believed this was an issue that was fixed a couple versions ago. What version of ember-cp-validations are you on?

I dont know where your `formData` is coming from or how its being handled so im gonna treat it as a black box. Here is a solution that should work 

```js
const Validations = buildValidations({
  'appName': {
    description: 'App Name',
    validators: [
      validator('presence', true)
    ]
  },
  'url': {
    description: 'URL',
    validators: [
      validator('presence', true),
      validator('format', {
        type: 'url',
        description: 'The URL'
      })
    ]
  }
});

export default Ember.Component.extend(Validations, {
  classNameBindings: ['styles.container'],
  formData: null,
  
  model: null, // This will hold our validate formData
  
  init() {
    this._super(...arguments);
     
    // Set model to an ember object with validations. Then pass it in the formData and the owner
    this.set('model', Ember.Object.extend(Validations).create(Ember.getOwner(this).ownerInjection(), this.get('formData')));

   // From here on, you can just do this.get('model').validate().then( do stuff here ) 
  }
});
```

Please note that the above example is just to help unblock you. If you can replicate this issue in a dummy app that I can get access to, I can help debug the source of the issue.

**Comment by offirgolan** [comment]
Im gonna go ahead and assume this issue has been resolved. Feel free to reopen if needed.

**Comment by wayneiny** [comment]
I have this issue as well. 

I'm doing some validations on an attribute of a model and using the validated-input example you provided in demo.

The thing is that after saving and setting the model to null, the validation is still running and it's causing this error
`ember.debug.js:15793Uncaught Error: Property set failed: object in path "model" could not be found or was destroyed.`

Is there a way to properly destroy it? Thanks in advance.

**Comment by offirgolan** [comment]
@wayneiny this is a fairly old thread. Can you create a new issue and give more details about the exact problem you are facing?

**Comment by lolmaus** [comment]
I store validations on an Ember Data model.

On submitting my form, I unload the saved record and replace it with a new one.

This was causing "this field is required" validation errors to appear.

I resolved the issue by wrapping `this.model = this.store.createRecord('foo')` with `next()`.

**Comment by lolmaus** [comment]
Alternatively, if you're using `ember-bootstrap`, the `hideValidationsOnSubmit` [option](hideValidationsOnSubmit ) can be enabled on the form.
